src/DragHandle.tsx

```diff
@@ -19,6 +23,108 @@ const cellDragHandle = css`
 
 const cellDragHandleClassname = `rdg-cell-drag-handle ${cellDragHandle}`;
 
-export default function DragHandle(props: React.HTMLAttributes<HTMLDivElement>) {
-  return <div className={cellDragHandleClassname} {...props} />;
+interface Props<R, SR> extends Pick<DataGridProps<R, SR>, 'rows' | 'onRowsChange' | 'onFill'> {
+  columns: readonly CalculatedColumn<R, SR>[];
+  selectedPosition: SelectCellState | EditCellState<R>;
+  draggedOverRowIdx: number | undefined;
+  setDragging: (isDragging: boolean) => void;
+  setDraggedOverRowIdx: (overRowIdx: number | undefined) => void;
+}
+
+export default function DragHandle<R, SR>({
+  rows,
+  columns,
+  selectedPosition,
+  draggedOverRowIdx,
+  onRowsChange,
+  onFill,
+  setDragging,
+  setDraggedOverRowIdx
+}: Props<R, SR>) {
+  const latestDraggedOverRowIdx = useRef(draggedOverRowIdx);
+  useEffect(() => {
```

Comment:
I think `useEffect` is fine here but we can pass the `latestDraggedOverRowIdx` ref instead if you think it is more safe

Reply:
Could just do
```ts
function handleDragEnd() {
  setDraggedOverRowIdx((draggedOverRowIdx) => {
    // ...
    return undefined;
  });
}
```